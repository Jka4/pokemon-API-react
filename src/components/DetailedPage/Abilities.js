import React from "react";

const Abilities = (props) => {
	const abilities = props?.abilities;

	return (
		<div className='abilities'>
			<ul className='abilitiesUL'>
				<div className='skills_title'>Abilities</div>

				{abilities && abilities.map((i, key) => (
					<li className='abilitiesLI' key={key}>
						{key + 1}: {i.ability.name}
					</li>
				))}
			</ul>
		</div>
	);
};

export default Abilities;