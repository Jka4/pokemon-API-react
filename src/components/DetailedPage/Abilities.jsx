import React from "react";
import Paper from "@material-ui/core/Paper";

const Abilities = ({ abilities = [] }) => {
  return (
    <div className="abilities">
      <Paper elevation={3} className="abilitiesUL">
        <div className="skills_title">Abilities</div>

        {abilities.map((element, key) => (
          <li className="abilitiesLI" key={key}>
            {key + 1}: {element.ability.name}
          </li>
        ))}
      </Paper>
    </div>
  );
};

export default Abilities;
