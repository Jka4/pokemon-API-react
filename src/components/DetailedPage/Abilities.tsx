import React from "react";
import Paper from "@material-ui/core/Paper";
import Skeleton, { SkeletonTheme } from 'react-loading-skeleton';

type Props = {
  abilities: AbilityElements[];
};

type AbilityElements = {
  ability: {
    url: string;
    name: string;
  };
}

const Abilities: React.FC<Props> = ({ abilities = [] }: Props) => {

  const fallbackSkeletons = () => {
    return (
      <SkeletonTheme color="#53aeff" highlightColor="#0066be">
        {[1, 2, 3, 4, 5, 6].map(el => (
          <p key={el} >
            <Skeleton height={15} width={100} />
          </p>
        ))}
      </SkeletonTheme>
    );
  };

  return (
    <div className="abilities">
      <Paper elevation={3} className="abilitiesUL">
        <div className="skills_title">Abilities</div>
        {abilities.length !== 0 ? (abilities.map((element: AbilityElements) => (
          <li className="abilitiesLI" key={element.ability.name}>
            - {element.ability.name}
          </li>
        ))) : (
            fallbackSkeletons()
          )}
      </Paper>
    </div>
  );
};

export default Abilities;
