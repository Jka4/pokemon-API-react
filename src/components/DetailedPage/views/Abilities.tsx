import React from 'react';
import styled from 'styled-components';
import { Skeleton } from '@material-ui/lab';

interface Props {
  abilities: AbilityElements[];
}

interface AbilityElements {
  ability: {
    url: string;
    name: string;
  };
}

const Abilities: React.FC<Props> = ({ abilities }: Props) => {
  const fallbackSkeletons = () => {
    const fakeArr = Array.from(Array(7).keys());

    return (
      <>
        {fakeArr.map((el) => (
          <LI key={el}>
            <Skeleton animation="wave" width={135} height={30} />
          </LI>
        ))}
      </>
    );
  };

  const abilitiesRow = () => {
    return (
      <>
        {abilities.map((element: AbilityElements) => (
          <LI key={element.ability.name}>- {element.ability.name}</LI>
        ))}
      </>
    );
  };

  const showSkeletons = abilities.length === 0;

  return (
    <AbilitiesStyled>
      <UL>
        <Title>Abilities</Title>
        {showSkeletons ? fallbackSkeletons() : abilitiesRow()}
      </UL>
    </AbilitiesStyled>
  );
};

const AbilitiesStyled = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  margin-left: -1rem;
  width: 200px;
  height: 380px;
`;

const Title = styled.div`
  align-self: center;
  justify-self: center;
  font-size: 2rem;

  @media only screen and (min-device-width: 320px) and (max-device-width: 480px) {
    font-size: 1.5rem;
  }
`;

const UL = styled.ul`
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  justify-content: flex-start;
  border: 1px solid black;
  border-radius: 0.7rem;
  background-color: #53aeff !important;
  padding: 25px;
  width: 100%;
  height: 100%;
  margin-top: 0;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;

  @media only screen and (min-device-width: 320px) and (max-device-width: 480px) {
    padding: 10px;
  }
`;

const LI = styled.li`
  display: flex;
  border-radius: 0.7rem;
  background-color: #53aeff !important;
  padding: 6px;
  width: 100%;
  font-size: 18px;
  text-transform: uppercase;

  @media only screen and (min-device-width: 320px) and (max-device-width: 480px) {
    font-size: 14px;
  }
`;

export default Abilities;
