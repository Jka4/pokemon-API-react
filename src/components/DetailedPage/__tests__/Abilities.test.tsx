import * as React from 'react';

import Abilities from '../views/Abilities';

import { mount, shallow } from 'enzyme';

describe('<Abilities/> ', () => {
  const fakeData = [
    {
      ability: {
        name: 'swarm',
        url: 'https://pokeapi.co/api/v2/ability/68/',
      },
    },
    {
      ability: {
        name: 'insomnia',
        url: 'https://pokeapi.co/api/v2/ability/15/',
      },
    },
  ];

  const container = mount(<Abilities abilities={fakeData} />);

  it('renders correctly', () => {
    expect(container).toMatchSnapshot();
  });
});
