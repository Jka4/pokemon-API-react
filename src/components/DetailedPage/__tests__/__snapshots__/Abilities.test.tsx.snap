// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Abilities/>  renders correctly 1`] = `
<Abilities
  abilities={
    Array [
      Object {
        "ability": Object {
          "name": "swarm",
          "url": "https://pokeapi.co/api/v2/ability/68/",
        },
      },
      Object {
        "ability": Object {
          "name": "insomnia",
          "url": "https://pokeapi.co/api/v2/ability/15/",
        },
      },
    ]
  }
>
  <styled.div>
    <div
      className="sc-gsTCUz lmOHhs"
    >
      <styled.ul>
        <ul
          className="sc-hKgILt fHTici"
        >
          <styled.div>
            <div
              className="sc-dlfnbm cgjAkB"
            >
              Abilities
            </div>
          </styled.div>
          <styled.li
            key="swarm"
          >
            <li
              className="sc-eCssSg jZMqxQ"
            >
              - 
              swarm
            </li>
          </styled.li>
          <styled.li
            key="insomnia"
          >
            <li
              className="sc-eCssSg jZMqxQ"
            >
              - 
              insomnia
            </li>
          </styled.li>
        </ul>
      </styled.ul>
    </div>
  </styled.div>
</Abilities>
`;
